radix-engine/src/system/system_modules/costing/fee_table.rs

```diff
@@ -399,16 +399,62 @@ impl FeeTable {
 
     #[inline]
     pub fn bls12381_v1_aggregate_verify_cost(&self, sizes: &[usize]) -> u32 {
-        // Below approach does not take aggregation into account.
-        // Summing costs pers size gives greater values.
-        // We've found somewhat difficult to find a proper and effective equation to estimate
-        // the instructions count collected with `test_crypto_scrypto_verify_bls12381_v1_costing`
-        // TODO: Find more adequate cost estimation
-        let mut cost: u32 = 0;
-        for size in sizes {
-            cost += self.bls12381_v1_verify_cost(*size);
+        // Observed that aggregated verify might be broken down into:
+        // - steps depending on message size
+        //   - aggregation of pairings of each corresponding key and message pair
+        //   - commit each above aggregation
+        // - steps that do not depend on message size
+        //   - read signature from bytes: 281125 instructions
+        //   - signature validation: 583573 instructions
+        //   - aggregated pairing of signature to verify and initialization point: 3027639 instructions
+        //   - final verification: 4280077 instructions
+        //
+        // more details and data in https://docs.google.com/spreadsheets/d/1rV0KyB7UQrg2tOenbh2MQ1fo9MXwrwPFW4l0_CVO-6o/edit?usp=sharing
+
+        // Pairing aggregate
+        // Following linerar equation might be applied:
+        //   (used: https://www.socscistatistics.com/tests/regression/default.aspx)
+        //   instructions_cnt = 34.42199 * size + 2620295.64271
+        //   Lets round:
+        //    34.42199      -> 35
+        //    2620295.64271 -> 2620296
+        //
+        // Also observed that additional 16850000 instructions are performed
+        // every multiple of 8
+        let mut instructions_cnt = 0u32;
+
+        // P
+        for s in sizes {
+            instructions_cnt = add(add(instructions_cnt, mul(35, cast(*s))), 2620296);
         }
-        cost
+        let multiplier = cast(sizes.len() / 8);
+
+        instructions_cnt = add(instructions_cnt, mul(multiplier, 16850000));
+
+        // Pairing commit
+        // Observed that number commit instructions repeats every multiple of 8
+        instructions_cnt = add(
+            instructions_cnt,
+            match sizes.len() % 8 {
+                1 => 3051556,
+                2 => 5020768,
+                3 => 6990111,
+                4 => 8959454,
+                5 => 10928798,
+                6 => 12898141,
+                7 => 14867484,
+                0 => 0,
```

Comment:
Can we move `0 => 0` to the first for readability?

Reply:
Done